src/types/parameters.ts

```diff
@@ -1,12 +1,35 @@
-import { NonNegativeInteger } from "@pagopa/ts-commons/lib/numbers";
 import * as t from "io-ts";
 import { NonEmptyString } from "@pagopa/ts-commons/lib/strings";
-import { withDefault } from "@pagopa/ts-commons/lib/types";
+import { NonNegativeIntegerFromString } from "@pagopa/ts-commons/lib/numbers";
+
+/**
+ * This codec is defined in ts-commons, but since ts-commons is at io-ts 1.8.x
+ * we cannot import because it generates an error on types.
+ *
+ * TODO: import from ts-commons when io-backend will be upgraded to new fp-ts and io-ts
+ */
+type BooleanFromString = t.Type<boolean, string, unknown>;
+
+const BooleanFromString: BooleanFromString = new t.Type<
+  boolean,
+  string,
+  unknown
+>(
+  "BooleanFromString",
+  t.boolean.is,
+  (s, c) =>
+    s === "true"
+      ? t.success(true)
+      : s === "false"
+      ? t.success(false)
+      : t.failure(s, c),
+  String
+);
 
 export const GetMessagesParameters = t.partial({
   /* eslint-disable sort-keys */
-  pageSize: withDefault(NonNegativeInteger, 100 as NonNegativeInteger),
-  enrichResultData: withDefault(t.boolean, false),
+  pageSize: NonNegativeIntegerFromString,
+  enrichResultData: BooleanFromString,
   maximumId: NonEmptyString,
   minimumId: NonEmptyString
```

Comment:
maybe we can think about try to refactor `api_backend.yaml` so we could generate it?

Reply:
I think generation does not work for this kind of params without giving an hard twist to the codegen api.

Reply:
I thought we could use our `x-import` attribute, like we do for [FiscalCode](https://github.com/pagopa/io-functions-commons/blob/e0f356fac126b4c1ae06afcbcf1b10881e1b4643/openapi/definitions.yaml#L359)

Reply:
Mmm interesting, we could try.

Reply:
Done in https://github.com/pagopa/io-backend/pull/835/commits/bb1abf36012977382a61cda88fd85919a13cb37c.
I made a new "booleans" module in "types" folder because BooleanFromString is only available on latest ts-commons package. This module will have to be removed when we will upgrade the backend to fp-ts 2.x and upgrade ts-commons.

Reply:
Amazing, good job! Thanks